evil-owl.el

```diff
@@ -213,22 +232,65 @@ MARK points nowhere."
                            ((markerp pos) (marker-buffer pos)))))
     (with-current-buffer buffer
       (let ((line (line-number-at-pos pos))
-            (column (save-excursion
-                      (goto-char pos)
-                      (current-column))))
-        (list line column (current-buffer))))))
+            (display-string "")
+            column)
+        (save-excursion
+          (goto-char pos)
+          (setq column (current-column))
+          (when evil-owl-mark-context-lines
+            (let ((lines-traversed-upward 0)
+                  display-string-beg
+                  display-string-end)
+              (while (and (< lines-traversed-upward evil-owl-mark-context-lines)
+                          (not (bobp)))
+                (forward-line -1)
+                (setq lines-traversed-upward (+ 1 lines-traversed-upward)))
+              (setq display-string-beg (line-beginning-position))
+              (dotimes (_ (+ lines-traversed-upward
+                             evil-owl-mark-context-lines))
+                (forward-line 1))
+              (setq display-string-end (line-end-position))
+              (setq display-string
+                    (evil-owl--format-mark-display-string
+                     (buffer-substring
+                      display-string-beg
+                      display-string-end))))))
```

Comment:
By the way, even if you choose to go back to having only one line of context, I just want to point out that `line-beginning-position` and `line-end-position` take counts, so all the `forward-line`s were unnecessary. Maybe a tip for the future :)